python/orca/tutorial/NCF/process_xshards.py

```diff
@@ -68,22 +68,48 @@ def split_dataset(df):
     return train_data, test_data
 
 
-def prepare_data(dataset_dir, num_ng=4):
+def prepare_data(data_dir="./", dataset="ml-1m", num_ng=4):
     print("Loading data...")
-    # Need spark3 to support delimiter with more than one character.
-    users = read_csv(
-        os.path.join(dataset_dir, "users.dat"),
-        sep="::", header=None, names=["user", "gender", "age", "occupation", "zipcode"],
-        usecols=[0, 1, 2, 3, 4],
-        dtype={0: np.int64, 1: str, 2: np.int32, 3: np.int64, 4: str})
-    ratings = read_csv(
-        os.path.join(dataset_dir, "ratings.dat"),
-        sep="::", header=None, names=["user", "item"],
-        usecols=[0, 1], dtype={0: np.int64, 1: np.int64})
-    items = read_csv(
-        os.path.join(dataset_dir, "movies.dat"),
-        sep="::", header=None, names=["item", "category"],
-        usecols=[0, 2], dtype={0: np.int64, 2: str})
+    if dataset == "ml-1m":
+        # Need spark3 to support delimiter with more than one character.
+        users = read_csv(
+            os.path.join(data_dir, dataset, "users.dat"),
+            sep="::", header=None, names=["user", "gender", "age", "occupation", "zipcode"],
+            usecols=[0, 1, 2, 3, 4],
+            dtype={0: np.int64, 1: str, 2: np.int32, 3: np.int64, 4: str})
+        ratings = read_csv(
+            os.path.join(data_dir, dataset, "ratings.dat"),
+            sep="::", header=None, names=["user", "item"],
+            usecols=[0, 1], dtype={0: np.int64, 1: np.int64})
+        items = read_csv(
+            os.path.join(data_dir, dataset, "movies.dat"),
+            sep="::", header=None, names=["item", "category"],
+            usecols=[0, 2], dtype={0: np.int64, 2: str})
+    else:  # dataset == "ml-100k"
+        users = read_csv(
+            os.path.join(data_dir, dataset, "u.user"),
+            sep="|", header=None, names=["user", "age", "gender", "occupation", "zipcode"],
+            usecols=[0, 1, 2, 3, 4],
+            dtype={0: np.int64, 1: np.int32, 2: str, 3: str, 4: str})
+        ratings = read_csv(
+            os.path.join(data_dir, dataset, "u.data"),
+            sep="\t", header=None, names=["user", "item"],
+            usecols=[0, 1], dtype={0: np.int64, 1: np.int64})
+        items = read_csv(
+            os.path.join(data_dir, dataset, "u.item"),
+            sep="|", header=None,
+            names=["item"]+[f"col{i}" for i in range(19)],
+            usecols=[0]+list(range(5, 24)),
+            dtype=str)
+
+        # merge multiple one-hot columns into one category column
+        def merge_one_hot_cols(df):
+            df["item"] = df["item"].astype(np.int64)
```

Comment:
why add this?

Reply:
![image](https://user-images.githubusercontent.com/42887453/220000263-f08d9f55-0455-4919-aff6-88b3eeafec0b.png)
The datatype of all columns is str when read items using `read_csv`. So the column `item` should be set to `int64` manually.

Reply:
okay, then what about using np.int32?

Reply:
or as you need to concat them to string, does it matter whether it is a str or int?